src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageCommand.cs

```diff
@@ -0,0 +1,168 @@
+            Func<IListPackageCommandRunner> getCommandRunner)
+        {
+            app.Command("list", listpkg =>
+            {
+                listpkg.Description = Strings.ListPkg_Description;
+                listpkg.HelpOption(XPlatUtility.HelpOption);
+
+                listpkg.Option(
+                    CommandConstants.ForceEnglishOutputOption,
+                    Strings.ForceEnglishOutput_Description,
+                    CommandOptionType.NoValue);
+
+                var path = listpkg.Argument(
+                    "<PROJECT | SOLUTION>",
+                    Strings.ListPkg_PathDescription,
+                    multipleValues: false);
+
+                var framework = listpkg.Option(
+                    "--framework",
+                    Strings.ListPkg_FrameworkDescription,
+                    CommandOptionType.SingleValue);
+
+                var outdated = listpkg.Option(
+                    "--outdated",
+                    Strings.ListPkg_OutdatedDescription,
+                    CommandOptionType.NoValue);
+
+                var deprecated = listpkg.Option(
+                    "--deprecated",
+                    Strings.ListPkg_DeprecatedDescription,
+                    CommandOptionType.NoValue);
+
+                var transitive = listpkg.Option(
+                    "--include-transitive",
+                    Strings.ListPkg_TransitiveDescription,
+                    CommandOptionType.NoValue);
+
+                var prerelease = listpkg.Option(
+                    "--include-prerelease",
+                    Strings.ListPkg_PrereleaseDescription,
+                    CommandOptionType.NoValue);
+
+                var highestPatch = listpkg.Option(
+                    "--highest-patch",
+                    Strings.ListPkg_HighestPatchDescription,
+                    CommandOptionType.NoValue);
+
+                var highestMinor = listpkg.Option(
+                    "--highest-minor",
+                    Strings.ListPkg_HighestMinorDescription,
+                    CommandOptionType.NoValue);
+
+                var source = listpkg.Option(
+                    "--source",
+                    Strings.ListPkg_SourceDescription,
+                    CommandOptionType.SingleValue);
+
+                var config = listpkg.Option(
+                    "--config",
+                    Strings.ListPkg_ConfigDescription,
+                    CommandOptionType.SingleValue);
+
+                listpkg.OnExecute(async () =>
+                {
+                    var logger = getLogger();
+
+                    var settings = ProcessConfigFile(config.Value(), path.Value);
+                    var sources = ParseArgumentWithCommas(source.Value());
+                    
+                    var packageSources = GetPackageSources(settings, sources, config);
+
+                    var frameworks = ParseArgumentWithCommas(framework.Value());
```

Comment:
ohh so we do support comma separated values for framework... then just pass that then

Reply:
It's allowed by writing `--source` multiple times for restore